lib/malloy/client/type_traits.hpp

```diff
@@ -7,6 +7,24 @@
 #include "malloy/type_traits.hpp"
 
 namespace malloy::client::concepts {
+    namespace detail
+    {
+        template<typename Callback>
```

Comment:
What is the purpose of this struct template?
Some doxygen might be helpful :)

Reply:
I'll add some docs, but it is to carry the type of the functor being checked by the concept. Effectively this is a lambda with capture `cb`, but capturing lambdas run into issues in concepts which are hard to diagnose (the expression doesn't compile so the concept is simply always unsatisfied)